src/bloqade/qasm2/passes/qasm2py.py

```diff
@@ -0,0 +1,55 @@
+"""Rewrite qasm dialects into py dialects."""
+
+from kirin import ir
+from kirin.passes import Pass
+from kirin.rewrite import Walk, Fixpoint
+from kirin.dialects import py, math
+from kirin.rewrite.abc import RewriteRule
+from kirin.rewrite.result import RewriteResult
+from bloqade.qasm2.dialects import core, expr
+
+
+class _QASM2Py(RewriteRule):
+    """Rewrite qasm dialects into py dialects."""
+
+    UNARY_OPS = {
+        expr.Neg: py.USub,
+        expr.Sin: math.sin,
+        expr.Cos: math.cos,
+        expr.Tan: math.tan,
+        expr.Exp: math.exp,
+        expr.Sqrt: math.sqrt,
+    }
+
+    BINARY_OPS = {
+        expr.Add: py.Add,
+        expr.Sub: py.Sub,
+        expr.Mul: py.Mult,
+        expr.Div: py.Div,
+        expr.Pow: py.Pow,
+    }
+
+    def rewrite_Statement(self, node: ir.Statement) -> RewriteResult:
```

Comment:
Are we forgoing the usual method dispatch technique (where you have `rewrite_Statement` just forward to smaller methods that specifically target certain arg types) going forward?

Reply:
No, this is a special case where you don't need to dispatch to different functions because they would all be the same one line function so instead we just simply construct the IR node in the if statement.